US Gun Deaths Project

In [9]:
##introducing US Gun Deaths Data
import csv
f = open("guns.csv")
csv = csv.reader(f)

data = list(csv)
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [16]:
##Removing Headers From A List Of Lists
headers = data[:1]
data = data[1:]
print(headers)
print(data[0:5])


[['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3']]
[['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2'], ['6', '2012', '02', 'Suicide', '0', 'M', '17', 'Native American/Native Alaskan', '100', 'Home', '1'], ['7', '2012', '02', 'Undetermined', '0', 'M', '48', 'White', '100', 'Home', '2']]


In [14]:
##Counting Gun Deaths By Year

"""
years = []
for row in data:
    years.append(row[1])
"""
#list the year
years = [row[1] for row in data]

#Count by the year
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year] += 1
    else:
        year_counts[year] = 1

year_counts


{'2012': 33563, '2013': 33636, '2014': 33599}

In [33]:
##Exploring Gun Deaths By Month And Year

import datetime

"""
dates = []
for row in data:
    year = int(row[1])
    month = int(row[2])
    day1 = datetime.datetime(year = year, month = month, day=1)
    dates.append(day1)
"""
#list the year and month
dates = [datetime.datetime(year = int(row[1]), month = int(row[2]), day = 1) for row in data]
print(dates[0:5])  

#count by the year and month
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1

#print(date_counts)
date_counts

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


{datetime.datetime(2012, 1, 1, 0, 0): 2756,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [36]:
##Exploring Gun Deaths By Race And Sex

sex_counts = {}
race_counts = {}

for row in data:
    sex = row[5]
    race = row[7]
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1
        
print(sex_counts)
print(race_counts)
    
    

{'M': 86348, 'F': 14448}
{'Black': 23296, 'Asian/Pacific Islander': 1325, 'Native American/Native Alaskan': 917, 'White': 66236, 'Hispanic': 9022}


Findings so far

by sex:
The number of gun death from men in the US is around 6 times than the number from women based the data during 2012 to 2014.    

by race:
The number of gun death from White is the highest(66236), which is almost 3 times than the number from Black (the second highest, 23296). The numbers following by Black are 9022(Hispanic), 1325(Asian/Pacific Islander) and 917(Native American/Native Alaskan). 

Since the total populations of each race are highly different, to examine the gun deaths by race rate would make more sense than counting the numbers purely. So the next section, the population numbers of each race would be imported for more insights.

In [64]:
###introducing US census Data
import csv
f = open("census.csv")
o = csv.reader(f)
census = list(o)
print(census)


[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [41]:
#Computing Rates Of Gun Deaths Per Race

#Coumputing total population by race
mapping = {
    "Asian/Pacific Islander":int(census[1][14]) + int(census[1][15]),
    "Black":int(census[1][12]),
    "Native American/Native Alaskan":int(census[1][13]),
    "Hispanic":int(census[1][11]),
    "White":int(census[1][10])    
}
print(mapping)

{'Black': 40250635, 'Asian/Pacific Islander': 15834141, 'Native American/Native Alaskan': 3739506, 'Hispanic': 44618105, 'White': 197318956}


In [43]:
race_per_hundredk = {}

for row in race_counts:
    race_per_hundredk[row] = (race_counts[row] / mapping[row]) * 100000

"""
for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000
"""

race_per_hundredk 

{'Asian/Pacific Islander': 8.36799419684339,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.567986240511026}

Findings so far

From last section, the number of gun death from White is the highest, however White's the gun death rate after comparing with the population is around 33.57%, which is lower than Black(57.88%). The lowest gun death rate is from Asian/Pacific Islander(8.37%), even they are the forth largest race in the US. The third largest race, Hispanic has the gun death rate of 20.22%, while the smallest race Native American/Native Alaskan has the rate of 24.52%. 


In [67]:
#examining by intent (homicide rate) and race

intents = [row[3] for row in data] #reasons of gun deaths
homicide_race_counts = {}
races = [row[7] for row in data]

for i,race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1

for row in homicide_race_counts:
    homicide_race_counts[row] = (homicide_race_counts[row] / mapping[row]) * 100000

"""
for k,v in homicide_race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000
"""
    
print(homicide_race_counts) # the homicide rate by race 

{'Black': 48.471284987180944, 'Asian/Pacific Islander': 3.530346230970155, 'Native American/Native Alaskan': 8.717729026240365, 'White': 4.6356417981453335, 'Hispanic': 12.627161104219914}


Findings so far

Black has the highest homicide rate(48.47%), following by Hispanic(12.63%), Native American/Native Alaskan(8.72%), White(4.64%) and Asian/Pacific Islander(3.53%).

In [68]:
#Examining homicide rate by month and gender
month_homi_count = {}
gender_homi_count = {}
gender_homi_rate = {}
#months = [row[2] for row in data]

for row in data:
    if int(row[2]) not in month_homi_count:
        month_homi_count[int(row[2])] = 0
    if row[5] not in gender_homi_count:
        gender_homi_count[row[5]] = 0
    if row[3] == "Homicide":
        month_homi_count[int(row[2])] += 1
        gender_homi_count[row[5]] += 1
        
print(month_homi_count)
print(gender_homi_count)

#Explore the homicide rate by gender.
for k,v in gender_homi_count.items():
    gender_homi_rate[k] = (v / sex_counts[k]) * 100
gender_homi_rate

{1: 2829, 2: 2178, 3: 2780, 4: 2845, 5: 2976, 6: 3130, 7: 3269, 8: 3125, 9: 2966, 10: 2968, 11: 2919, 12: 3191}
{'M': 29803, 'F': 5373}


{'F': 37.188538205980066, 'M': 10.350420226436064}

Findings so far

By month: Homicide rates have no obvious differences between each month, but the higher numbers are located in Summer(June, July and August) and December. The lowest number is in February.

By gender: Male has almost 4 times of the homicide number(29803) than female(5373). However, female's homicide rate(37.19%) is distinctly higher than male(10.35%).

In [74]:
#Explore the rates of other intents, like Accidental, by gender and race.
all_intents = set(intents) #unique intents

for row in data:
    if row[3] not in all_intents_count:
        all_intents_count[row[3]] = 0
    all_intents_count[row[3]] += 1
all_intents_count    


{'Accidental': 4917,
 'Homicide': 105528,
 'NA': 3,
 'Suicide': 189519,
 'Undetermined': 2421}

In [ ]:
#Here are some potential next steps:

#Figure out the link, if any, between month and homicide rate.
#Explore the homicide rate by gender.

#not done yet
####Explore the rates of other intents, like Accidental, by gender and race.
####Find out if gun death rates correlate to location and education.